typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,541 @@
+                }
+            })
+        });
+        const resultLoadLambda: Result<any, Error> = await promiseLoadLambda;
+        if (!resultLoadLambda.isOk()) {
+            return err(resultLoadLambda.error);
+        }
+
+        const moduleCtx: LambdaContext = new LambdaContext(this.client, lambdaId);
+        return ok(moduleCtx);
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async getLambdaContext(lambdaId: LambdaID): Promise<Result<LambdaContext, Error>> {
+        const args: GetLambdaInfoArgs = newLambdaInfoArgs(lambdaId);
+        
+        const promiseGetLambdaInfo: Promise<ResultAsync<GetLambdaInfoResponse, Error>> = new Promise((resolve, _unusedReject) => {
+            this.client.getLambdaInfo(args, (error: grpc.ServiceError, response: GetLambdaInfoResponse) => {
+                if (error) {
+                    resolve(errAsync(error));
+                } else {
+                    resolve(okAsync(response));
+                }
+            })
+        });
+        const resultGetLambdaInfo: Result<GetLambdaInfoResponse, Error> = await promiseGetLambdaInfo;
+        if (!resultGetLambdaInfo.isOk()) {
+            return err(resultGetLambdaInfo.error);
+        }
+
+        const lambdaCtx: LambdaContext = new LambdaContext(this.client, lambdaId);
+        return ok(lambdaCtx);
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async registerStaticFiles(staticFileFilepaths: Map<StaticFileID, string>): Promise<Result<null, Error>> {
+        const strSet: Map<string, boolean> = new Map();
+        for (let [staticFileId, srcAbsFilepath] of staticFileFilepaths.entries()) {
+            
+            // Sanity-check that the source filepath exists
+            const promiseSrcAbsFilepath: Promise<fs.Stats> = fsPromises.stat(srcAbsFilepath);
+            promiseSrcAbsFilepath.catch(() => { 
+                return err(new Error("Source filepath " + srcAbsFilepath + " associated with static file " + staticFileId + " doesn't exist")); 
+            })
+            
+            strSet[String(staticFileId)] = true;
+        }
+
+        const args: RegisterStaticFilesArgs = newRegisterStaticFilesArgs(strSet);
+        
+        const promiseRegisterStaticFiles: Promise<ResultAsync<RegisterStaticFilesResponse, Error>> = new Promise((resolve, _unusedReject) => {
+            this.client.registerStaticFiles(args, (error: grpc.ServiceError, response: RegisterStaticFilesResponse) => {
+                if (error) {
+                    resolve(errAsync(error));
+                } else {
+                    resolve(okAsync(response));
+                }
+            })
+        });
+        const resultRegisterStaticFiles: Result<RegisterStaticFilesResponse, Error> = await promiseRegisterStaticFiles;
+        if (!resultRegisterStaticFiles.isOk()) {
+            return err(resultRegisterStaticFiles.error);
```

Comment:
Question: this function returns a `Promise`. do you need this to be `errAsync`, or does it not matter?

Reply:
It actually works with either or. The reason why I decided to use `okAsync` and `errAsync` is because the documentation mentioned that they were meant for asynchronous tasks. Since our work with the promise is asynchronous, that is why I decided to use these commands instead of `ok` and `err`. Here is the documentation for your reference: https://github.com/supermacro/neverthrow#resultunwrapor-method